planning/work-items/usability-and-design.md

```diff
@@ -205,6 +205,360 @@ In this case, the Thing has enough resources and contains its own HTTP server.
+                {
+                    "op":["readproperty"],
+                    "contentType":"application/cbor",
+                    "href": "myDevice/properties/prop2",
+                    "htv:methodName":"GET"
+                },
+                {
+                    "op":"writeproperty",
+                    "contentType":"application/cbor",
+                    "href": "myDevice/properties/prop2"
+                    "htv:methodName":"POST"
+                }
+            ]
+        }
+    }
+}
+```
+
+With the second TD (v1.1), we can see that it got longer due to not using the defaults of the binding. This effect would be amplified with the number of the affordances and forms the TD has.
+
+##### Luca - Form-reuse + Connection descriptors
+
+Design ideas for Form-reuse:
+
+- `Form` gains two additional fields: `reusable` and `base`.
+- `Thing` gains an additional field: `bases` (`connections` in Ege's proposal).
+- `Thing::bases` is a container of `Form defaults`, a dictionary of Forms.
+- The `base` field is a string matching one of the keys of `Thing::bases`.
+- There is a single-inheritance pattern given by `Form::base`
+  - you can use the fields in the `Form` referred in `Form::base` as default for all the fields not present in the current `Form`
+  - `href` is additive; if present, it must be added to the `href` provided by the `base` (see normal approach to adding URIs)
+  - if `base` is set in the Form referred to by the current Form `base`, the base form has to be resolved first.
+    - Implementation-wise, all the Forms in `Thing::bases` can be resolved once and the result cached.
+  - All the other fields simply replace the default if they exist in the current Form
+- The `reusable` field is a boolean
+- A Consumer MAY decide to keep a connection alive and reuse it if `Form::reusable` is true; the exact behaviour is protocol-dependent
+  - if `Form::reusable` is set to false, the Consumer MUST tear down the connection and start anew.
+
+This make it simpler to reuse Forms, and provides the minimum amount of information to tell if a Consumer can reuse a connection.
+
+Design ideas for Connection descriptors
+
+- `Form` loses `subprotocol`
+- `Form` gains `protocol`
+- `Thing` gains `protocols` and `connections`
+- `Thing::connections` is a dictionary
+  - The key is a string
+  - The value is an Object with the following fields
+    - `href` providing the connection path if applies
+    - Any protocol-specific fields to be used to describe the connection
+- `Thing::protocols` is a dictionary
+  - The key is a string
+  - The value is a protocol identifier or an Object with the following fields
+    - A mandatory `protocol` containing a protocol identifier
+    - Additional protocol-specific fields
+- `Form::protocol` contains a string matching a key in `Thing::protocols`
+- `Form::connection` contains a string matching a key in `Thing::connections`
+
+Open questions:
+
+- Can we move the security machinery to connection?
```

Comment:
This item hadn't been discussed yet, it boils down to either
- remove `security` and related terms and use `Form` as container for what we have in the securityDefinitions. (less tidy IMHO)
- rename `security` to `connection` and `securityDefinitions` to `connectionDefinitions` and use it to contain the connection setup terms.

``` json
"connectionDefinitions": { "no_sec+http3": {
    "scheme": "nosec",
    "htv:version": "3",
    "transport": "quick",
    "tls:cyphers": "...
  }
}
```

```json
"bases": {
  "my_read_only_http3_properties": {
    "op": "readproperty",
    "connection": "no_sec+http3",
    "htv:methodName": "...
  } 
}
```

The reason why is that `security` is just a specialized kind of connection setup covering from trasport (tls) to application (oauth).